### PR TITLE
Fix bug for recovering a prepared but not committed txn

### PR DESCRIPTION
**Pull request by jowlyzhang**
This PR fix a bug for recovering a prepared Transaction that can contain user-defined timestamps.

The `Transaction::Put` type of APIs expect the key provided to be user key without timestamps. When the original transaction added a key for a column family that enables user-defined timestamps, say of size 8. Internally `WriteBatch::Put` will leave a placeholder 8 bytes for the final commit timestamp. For example:
https://github.com/facebook/rocksdb/blob/cec28aa90f0e38666c0b3485d197ecbe0c2a025f/db/write_batch.cc#L937

When rebuilding this transaction from a `WriteBatch` from WAL log, we should consider this and remove the tailing 8 bytes of a key before adding it via the public Transaction write APIs. 

Test plan:
Added unit test that would fail without this fix

### CODE REVIEW

File: utilities/transactions/write_committed_transaction_ts_test.cc [hunk 1]
```diff
@@ -314,6 +327,9 @@ TEST_P(WriteCommittedTxnWithTsTest, RecoverFromWal) {
     s = GetFromDb(ReadOptions(), handles_[1], "key1", /*ts=*/24, &value);
     ASSERT_OK(s);
     ASSERT_EQ("value_3", value);
+
+    s = GetFromDb(ReadOptions(), handles_[1], "baz", /*ts=*/24, &value);
```
Comment (cbi42) [inline utilities/transactions/write_committed_transaction_ts_test.cc:331]: Do you mean to also commit this transaction?

**Review by cbi42 — APPROVED** [review]
LGTM! If we have stress test with ts + txn, we could probably catch this.

File: utilities/transactions/write_committed_transaction_ts_test.cc (hunk 1 above)
Comment (jowlyzhang) [reply utilities/transactions/write_committed_transaction_ts_test.cc:331]: oh, sorry for the confusion. This is to test the case that a recovered prepared transaction is not commited, and the key should not be found. It's kind of what the "foo" key in the original test does, I kind of shifted it down to the last one.

**Comment by jowlyzhang** [comment]
> LGTM! If we have stress test with ts + txn, we could probably catch this.

Yep, I will try to fill this gap this half.

**Comment by jowlyzhang** [comment]
@cbi42 thank you for the quick review!

**Pull request closed without merging**